pkg/kubelet/dockershim/docker_image.go

```diff
@@ -33,8 +34,9 @@ import (
 func (ds *dockerService) ListImages(filter *runtimeapi.ImageFilter) ([]*runtimeapi.Image, error) {
 	opts := dockertypes.ImageListOptions{}
 	if filter != nil {
-		if imgSpec := filter.GetImage(); imgSpec != nil {
-			opts.Filters.Add("reference", imgSpec.Image)
+		if filter.GetImage().GetImage() != "" {
+			opts.Filters = dockerfilters.NewArgs()
```

Comment:
Why do we need to call this now, but not before? Is this a bug as well?

Reply:
No, just for allocating it only when required.